engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -49,6 +63,44 @@ def verify_product_activation
           raise ActionController::TranslatedError.new('Unexpected instance verification error has occurred')
         end
 
+        def update_cache(product_id)
+          cache_key = [request.remote_ip, @system.login, product_id].join('-')
+          # caches verification result to be used by zypper auth plugin
+          Rails.cache.write(cache_key, true, expires_in: 20.minutes)
+        end
+
+        def prepare_scc_request(uri_path, product)
+          # check registration code agains SCC request
+          # GET /subscriptions/products
+          header = {
+            'accept' => 'application/json,application/vnd.scc.suse.com.v4+json',
+            'Content-Type' => 'application/json',
+            'Authorization' => "Token token=#{params[:token]}"
+	  }
+	  scc_request = Net::HTTP::Get.new(uri_path, header)
+	  scc_request.body = {
+            identifier: product.identifier,
+            version: product.version,
+            arch: product.arch,
+            release_type: product.release_type,
+          }.to_json
+          scc_request
+        end
+
+        def scc_check_regcode(product)
+          # check that the regcode can access that product
+	  uri = URI.parse(SUBSCRIPTIONS_PRODUCTS_URL)
+          http = Net::HTTP.new(uri.host, uri.port)
+          http.use_ssl = true
+          scc_request = prepare_scc_request(uri.path, product)
+          response = http.request(scc_request)
+          logger.info(
+            "Response code is #{response.code} for the attempt to " \
+              "check registration code for #{product.product_string} to SCC"
+          )
+          raise InstanceVerification::Exception.new('No subscription with this Registration Code found') unless response.code_type == Net::HTTPOK
+        end
+
```

Comment:
Can those methods also move to the `scc_proxy` engine? They seem unrelated to instance verification.

Reply:
Yes, that is the idea, do a refactor after merging this PR